src/mpi/coll/allreduce/allreduce_intra_tree.c

```diff
@@ -70,13 +70,42 @@ int MPIR_Allreduce_intra_tree(const void *sendbuf,
             MPIR_Treealgo_tree_create_topo_aware(comm_ptr, tree_type, k, root,
                                                  MPIR_CVAR_ALLREDUCE_TOPO_REORDER_ENABLE, &my_tree);
     } else if (tree_type == MPIR_TREE_TYPE_TOPOLOGY_WAVE) {
+        MPIR_Csel_coll_sig_s coll_sig = {
+            .coll_type = MPIR_CSEL_COLL_TYPE__ALLREDUCE,
+            .comm_ptr = comm_ptr,
+            .u.allreduce.sendbuf = sendbuf,
+            .u.allreduce.recvbuf = recvbuf,
+            .u.allreduce.count = count,
+            .u.allreduce.datatype = datatype,
+            .u.allreduce.op = op,
+        };
+
+        MPII_Csel_container_s *cnt = MPIR_Csel_search(comm_ptr->csel_comm, coll_sig);
+        MPIR_Assert(cnt);
```

Comment:
How do we know that `MPIR_Csel_search` will return a tree algorithm container?

Reply:
ping @dycz0fx

Reply:
Fixed. Added a check (cnt->id == MPII_CSEL_CONTAINER_TYPE__ALGORITHM__MPIR_Allreduce_intra_tree) to make sure the MPIR_Csel_serach will return a tree algorithm container.